eradiate/radprops/rad_profile.py

```diff
@@ -559,105 +512,66 @@ def default_absorption_data_set(wavelength):
         )
         return data.open(category="absorption_spectrum", id=dataset_id)
 
-    def compute_sigma_a(self):
+    def eval_thermoprops_profile(self):
         """
-        Compute absorption coefficient.
+        Evaluate thermophysical properties.
         """
-        mode = eradiate.mode()
-        if not mode.is_monochromatic():
-            raise ModeError(f"unsupported mode {mode.id}")
-        else:
-            wavelength = mode.wavelength
-            self._wavelength = wavelength
-        if self._thermoprops is None:
-            self._thermoprops = self.compute_thermoprops()
+        return us76.make_profile(self.levels)
+
+    def eval_sigma_a(self, spectral_ctx):
+        """
+        Evaluate absorption coefficient given spectral context.
+        """
+        wavelength = spectral_ctx.wavelength
```

Comment:
We don't check `if eradiate.mode().is_monochromatic()` here (and other eval* functions)?

Reply:
We should, it's fixed in the spectral loop branch.